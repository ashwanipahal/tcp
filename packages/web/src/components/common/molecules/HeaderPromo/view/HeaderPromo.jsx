import React from 'react';
import PropTypes from 'prop-types';
import Row from '@tcp/core/src/components/common/atoms/Row';
import Col from '@tcp/core/src/components/common/atoms/Col';
import Image from '@tcp/core/src/components/common/atoms/Image';
import Carousel from '@tcp/core/src/components/common/molecules/Carousel';
import CarouselConfig from '@tcp/web/src/config/carousel';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import headerPromoStyles from '../HeaderPromo.style';
import { getIconPath } from '../../../../../utils';

const carouselConfig = CarouselConfig.PROMO_AREA_DEFAULTS;

const HeaderPromo = ({ className, dataPromo, mobile }) => {
  const wrapperClass = mobile ? 'header-promo-area--mobile' : 'header-promo-area--desktop';

  return (
    <div className={`header-promo__container ${className}`}>
      {mobile && (
        <Carousel options={carouselConfig} carouselTheme="dark" className={wrapperClass}>
          {dataPromo.map(promoItem => (
            <div className="header-promo__item">
              <div className={`header-promo-item__icon ${promoItem.linkClass.class}`}>
                <Image src={getIconPath(promoItem.linkClass.class)} alt="promotion" />
              </div>
              <div className="header-promo-item__content">
                <span
                  className="styled-text"
                  style={{ color: promoItem.textLines[0].color }}
                  contentEditable="true"
                  dangerouslySetInnerHTML={{ __html: promoItem.textLines[0].text }}
                />
                <span
                  style={{ color: promoItem.textLines[1].color }}
                  dangerouslySetInnerHTML={{ __html: promoItem.textLines[1].text }}
                />
              </div>
            </div>
          ))}
        </Carousel>
      )}
      {!mobile && (
        <Row centered className={wrapperClass}>
          {dataPromo.map(promoItem => (
            <Col
              colSize={{
                large: 4,
                medium: 8,
                small: 6,
              }}
            >
              <div className="header-promo__item">
                <div className={`header-promo-item__icon ${promoItem.linkClass.class}`}>
                  <Image src={getIconPath(promoItem.linkClass.class)} alt="promotion" />
                </div>
                <div className="header-promo-item__content">
                  <span
                    className="styled-text"
                    style={{ color: promoItem.textLines[0].color }}
                    contentEditable="true"
                    dangerouslySetInnerHTML={{ __html: promoItem.textLines[0].text }}
                  />
                  <span
                    style={{ color: promoItem.textLines[1].color }}
                    dangerouslySetInnerHTML={{ __html: promoItem.textLines[1].text }}
                  />
                </div>
              </div>
            </Col>
          ))}
        </Row>
      )}
    </div>
  );
};

HeaderPromo.propTypes = {
  className: PropTypes.string.isRequired,
  dataPromo: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number]))
    .isRequired,
  mobile: PropTypes.bool.isRequired,
};

export default withStyles(HeaderPromo, headerPromoStyles);
export { HeaderPromo as HeaderPromoVanilla };
