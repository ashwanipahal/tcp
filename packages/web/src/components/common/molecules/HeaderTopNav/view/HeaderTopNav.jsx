import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';

import { BrandTabs, PromotionalArea } from '../..';
import HeaderTopNavStyle from '../HeaderTopNav.style';

const HeaderTopNav = ({ className, brandTabs, promoMessageWrapper }) => {
  return (
    <div className={className}>
      <PromotionalArea mobile data={promoMessageWrapper} />
      <div className="header-topnav__row">
        <div className="header-topnav__brand-tabs">
          <BrandTabs data={brandTabs} />
        </div>
        <div className="header-topnav__promo-area">
          <PromotionalArea mobile={false} data={promoMessageWrapper} />
        </div>
        <div className="header-topnav__track-order">Track order</div>
      </div>
    </div>
  );
};

HeaderTopNav.propTypes = {
  className: PropTypes.string.isRequired,
  brandTabs: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  promoMessageWrapper: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default withStyles(HeaderTopNav, HeaderTopNavStyle);
export { HeaderTopNav as HeaderTopNavVanilla };
