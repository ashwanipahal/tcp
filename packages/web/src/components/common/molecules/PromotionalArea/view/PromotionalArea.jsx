import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import Carousel from '@tcp/core/src/components/common/molecules/Carousel';
import RichText from '@tcp/core/src/components/common/atoms/RichText';
import CarouselConfig from '@tcp/web/src/config/carousel';
import { getLocator } from '@tcp/web/src/utils';
import style from '../PromotionalArea.style';

const PromotionalArea = ({ className, data, mobile }) => {
  const carouselConfig = mobile
    ? CarouselConfig.CAROUSEL_OPTIONS
    : CarouselConfig.CAROUSEL_FADE_OPTIONS;
  const wrapperClass = mobile
    ? 'header-topnav__promo-area--mobile'
    : 'header-topnav__promo-area--tablet';

  return (
    <div className={className}>
      <div className={wrapperClass}>
        <Carousel options={carouselConfig} carouselTheme="dark">
          {data.map(promotion => {
            return (
              <RichText
                key={promotion.id}
                richTextHtml={promotion.text}
                dataLocator={getLocator('global_promoareaimg')}
              />
            );
          })}
        </Carousel>
      </div>
    </div>
  );
};

PromotionalArea.propTypes = {
  className: PropTypes.string.isRequired,
  data: PropTypes.arrayOf(
    PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string]))
  ).isRequired,
  mobile: PropTypes.bool,
};

PromotionalArea.defaultProps = {
  mobile: PropTypes.bool,
};

export default withStyles(PromotionalArea, style);
export { PromotionalArea as PromotionalAreaVanilla };
