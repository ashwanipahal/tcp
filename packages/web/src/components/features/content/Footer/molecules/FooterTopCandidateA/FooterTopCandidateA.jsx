import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  RichText,
  Col,
  Row,
  BodyCopy,
  TextItems,
} from '@tcp/core/src/components/common/atoms';
import { reduxForm } from 'redux-form';
import { Grid } from '@tcp/core/src/components/common/molecules';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { formatPhoneNumber } from '@tcp/core/src/utils/formValidation/phoneNumber';
import { getLocator } from '@tcp/core/src/utils';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';

import SocialMediaLinks from '../SocialMediaLinks';
/* TODO move to component itself */
import style from '../../Footer.style';

import FooterTopSignUpForm from '../FooterTopSignUpForm';

const emailSignupFieldName = 'signup';
const emailSignupSecondBrand = 'isEmailOptInSecondBrand';
const FooterTopEmailSignUpForm = reduxForm({
  form: 'FooterTopEmailSignUpForm', // a unique identifier for this form
  initialValues: {
    [emailSignupFieldName]: '',
    [emailSignupSecondBrand]: false,
  },
})(FooterTopSignUpForm);

const smsSignupFieldName = 'footerTopSmsSignup';
const textSignupSecondBrand = 'isTextOptInSecondBrand';
const FooterTopSmsSignUpForm = reduxForm({
  form: 'FooterTopSmsSignUpForm', // a unique identifier for this form
  initialValues: {
    [smsSignupFieldName]: '',
    [textSignupSecondBrand]: false,
  },
})(FooterTopSignUpForm);

class FooterTopCandidateA extends React.PureComponent {
  render() {
    const {
      emailSignup,
      emailSignupLabels,
      smsSignup,
      smsSignupLabels,
      socialMediaLinks,
      referAFriendButtonLabels,
      referAFriend,
      validateSignupEmail,
      validateSignupSmsPhoneNumber,
      submitEmailSubscription,
      submitSmsSubscription,
      emailSubscription,
      smsSubscription,
      openEmailSignUpModal,
      openSmsSignUpModal,
      isNavigationFooter,
    } = this.props;

    return (
      <Grid className="footer_top_candidate_a content-wrapper">
        <Row>
          {/* ------------ Email Sign Up starts here ----------------- */}
          <Col
            className="col-md-half-width"
            colSize={{
              large: 4,
              medium: 4,
              small: 6,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <BodyCopy
              component="h2"
              fontWeight="black"
              fontSize="fs15"
              className="heading_text email-sign-up"
              dataLocator="email_promo_text"
            >
              <TextItems textItems={emailSignup.textItems} />
            </BodyCopy>
            <FooterTopEmailSignUpForm
              labels={emailSignupLabels}
              validateForm={validateSignupEmail}
              onFormSubmit={submitEmailSubscription}
              subscription={emailSubscription}
              openSuccessModal={openEmailSignUpModal}
              dataLocators={{
                submitButton: 'email_submit_btn',
                inputField: 'enter_email_text_field',
                errorDataLocator: 'email_error_message',
                checkBox_gym: 'check_box_gym_opt_in',
                checkBox_tcp: 'check_box_tcp_opt_in',
              }}
              fieldName={emailSignupFieldName}
              secondFieldName={emailSignupSecondBrand}
            />

            <BodyCopy fontFamily="secondary" textAlign="center" fontSize={['fs9', 'fs9', 'fs12']}>
              <RichText richTextHtml={emailSignupLabels.lbl_SignUp_termsTextLabel} />
            </BodyCopy>
            <div>
              <div className="divider hide-in-medium-up" />
            </div>
          </Col>

          {/* ---------- Email Signup ends here ---------- */}

          {/* ---------- SMS Sign Up starts here ---------- */}
          <Col
            className="col-md-half-width"
            colSize={{
              large: 4,
              medium: 4,
              small: 6,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <BodyCopy
              component="h2"
              fontWeight="black"
              fontSize="fs15"
              className="heading_text sms_sign_up"
              dataLocator="sms_promo_text"
            >
              <TextItems textItems={smsSignup.textItems} />
            </BodyCopy>
            <FooterTopSmsSignUpForm
              labels={smsSignupLabels}
              fieldName={smsSignupFieldName}
              fieldProps={{
                normalize: formatPhoneNumber,
              }}
              validateForm={validateSignupSmsPhoneNumber}
              onFormSubmit={submitSmsSubscription}
              subscription={smsSubscription}
              openSuccessModal={openSmsSignUpModal}
              dataLocators={{
                submitButton: 'sms_submit_btn',
                inputField: 'sms_field',
                errorDataLocator: 'sms_error_message',
                checkBox_gym: 'sms_gym_opt_in',
                checkBox_tcp: 'sms_tcp_opt_in',
              }}
              secondFieldName={textSignupSecondBrand}
            />

            <BodyCopy fontFamily="secondary" textAlign="center" fontSize={['fs9', 'fs9', 'fs12']}>
              <RichText richTextHtml={smsSignupLabels.lbl_SignUp_termsTextLabel} />
            </BodyCopy>
          </Col>
          <div className="divider hide-in-large-up" />
          {/* ---------- SMS Signup ends here ------------ */}

          {/* ---------- Refer a friend start here-------- */}

          <Col
            className="refer_a_friend_desktop col-md-half-width"
            colSize={{
              large: 4,
              medium: 4,
              small: 6,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <Grid>
              <Row fullBleed className="flex-align-center">
                <Col
                  className="col-md-half-width"
                  colSize={{
                    large: 6,
                    medium: 4,
                    small: 7,
                  }}
                  ignoreGutter={{
                    small: true,
                  }}
                >
                  <BodyCopy
                    component="h2"
                    fontWeight="black"
                    fontSize="fs15"
                    className="heading_text refer-a-friend"
                  >
                    <TextItems textItems={referAFriend.textItems} />
                  </BodyCopy>
                </Col>
                <Col
                  colSize={{
                    large: 6,
                    medium: 4,
                    small: 6,
                  }}
                  ignoreGutter={{
                    small: false,
                  }}
                  className="candidate_a_inline_container_button col-md-half-width"
                >
                  <ClickTracker
                    as={Button}
                    id={
                      isNavigationFooter
                        ? 'extole_zone_global_navigation_footer'
                        : 'extole_zone_global_footer'
                    }
                    buttonVariation="variable-width"
                    data-locator={getLocator('refer_friend')}
                    className="refer_a_friend_button"
                    clickData={{
                      customEvents: ['event22', 'event80'],
                      pageShortName: 'content:referafriend  confirmation',
                      pageName: 'content:email confirmation',
                    }}
                  >
                    {referAFriendButtonLabels.text}
                  </ClickTracker>
                </Col>
              </Row>
              <div className="divider hide-in-medium-down" />
              <Row fullBleed className="hide-in-medium-down">
                <Col
                  className=""
                  colSize={{
                    large: 12,
                    medium: 12,
                    small: 12,
                  }}
                  ignoreGutter={{
                    small: true,
                  }}
                >
                  <div className="divider hide-in-medium-up" />
                  <SocialMediaLinks
                    {...socialMediaLinks}
                    className="footer_top_candidate_a_social_links"
                  />
                </Col>
              </Row>
            </Grid>
          </Col>
          {/* ---------- Refer a friend ends here-------- */}
          <div className="divider hide-in-medium-up" />
          <Col
            className="hide-in-large-up refer_a_frient_last_colm col-md-half-width"
            colSize={{
              large: 0,
              medium: 4,
              small: 12,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <SocialMediaLinks {...socialMediaLinks} className="social-media-links" />
          </Col>
        </Row>
      </Grid>
    );
  }
}

FooterTopCandidateA.propTypes = {
  className: PropTypes.string.isRequired,
  emailSignup: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  smsSignup: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  emailSignupLabels: PropTypes.shape({
    lbl_SignUp_placeholderText: PropTypes.string,
    lbl_SignUp_validationErrorLabel: PropTypes.string,
    lbl_SignUp_termsTextLabel: PropTypes.string,
    lbl_SignUp_submitButtonLabel: PropTypes.string,
    lbl_SignUp_gymSignUpLabel: PropTypes.string,
    lbl_SignUp_tcpSignUpLabel: PropTypes.string,
  }),
  smsSignupLabels: PropTypes.shape({
    lbl_SignUp_placeholderText: PropTypes.string,
    lbl_SignUp_validationErrorLabel: PropTypes.string,
    lbl_SignUp_termsTextLabel: PropTypes.string,
    lbl_SignUp_submitButtonLabel: PropTypes.string,
  }),
  socialMediaLinks: PropTypes.arrayOf(
    PropTypes.shape({
      icon_class: PropTypes.string,
      title: PropTypes.string,
    })
  ).isRequired,
  referAFriend: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  referAFriendButtonLabels: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  showError: PropTypes.bool,
  isEmailValid: PropTypes.bool,
  validateSignupEmail: PropTypes.func,
  validateSignupSmsPhoneNumber: PropTypes.func,
  submitEmailSubscription: PropTypes.func,
  submitSmsSubscription: PropTypes.func,
  openEmailSignUpModal: PropTypes.func,
  openSmsSignUpModal: PropTypes.func,
  emailSubscription: PropTypes.shape({}),
  smsSubscription: PropTypes.shape({}),
  isNavigationFooter: PropTypes.bool,
};

FooterTopCandidateA.defaultProps = {
  emailSignup: {
    text: '',
    title: '',
  },
  smsSignup: {
    text: '',
    title: '',
  },
  referAFriendButtonLabels: {
    title: '',
    text: '',
  },
  emailSignupLabels: {
    lbl_SignUp_placeholderText: '',
    lbl_SignUp_validationErrorLabel: '',
    lbl_SignUp_termsTextLabel: '',
    lbl_SignUp_submitButtonLabel: '',
    lbl_SignUp_gymSignUpLabel: '',
    lbl_SignUp_tcpSignUpLabel: '',
  },
  smsSignupLabels: {
    lbl_SignUp_placeholderText: '',
    lbl_SignUp_validationErrorLabel: '',
    lbl_SignUp_termsTextLabel: '',
    lbl_SignUp_submitButtonLabel: '',
  },
  referAFriend: {
    title: '',
    text: '',
  },
  showError: false,
  isEmailValid: false,
  validateSignupEmail: () => Promise.resolve(),
  validateSignupSmsPhoneNumber: () => Promise.resolve(),
  submitEmailSubscription: () => {},
  submitSmsSubscription: () => {},
  openEmailSignUpModal: () => {},
  openSmsSignUpModal: () => {},
  emailSubscription: {},
  smsSubscription: {},
  isNavigationFooter: false,
};

export default withStyles(FooterTopCandidateA, style);

export { FooterTopCandidateA as FooterTopCandidateAVanilla };
