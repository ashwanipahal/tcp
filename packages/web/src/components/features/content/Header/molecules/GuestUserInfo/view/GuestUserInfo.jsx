import React from 'react';
import PropTypes from 'prop-types';
import { Image, Button, BodyCopy } from '@tcp/core/src/components/common/atoms';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';
import { getIconPath, isCanada } from '@tcp/core/src/utils';
import ACCOUNT_CONSTANTS from '@tcp/core/src/components/features/account/Account/Account.constants';

const AnalyticsNavigationText = ACCOUNT_CONSTANTS.ACCOUNT_ANALYTICS.navigationText;

const handleUserRewards = userRewards => {
  return userRewards % 1 ? userRewards : Math.floor(userRewards);
};

const GuestUserInfo = ({
  createAccount,
  login,
  openOverlay,
  triggerLoginCreateAccount,
  onLinkClick,
  userNameClick,
  isRememberedUser,
  userName,
  isDrawer,
  userPoints,
  userRewards,
}) => {
  const LoginLinkClick = e =>
    onLinkClick(
      {
        e,
        openOverlay,
        userNameClick,
        triggerLoginCreateAccount,
        navname: AnalyticsNavigationText.logIn,
      },
      login
    );

  return (
    <React.Fragment>
      {!isRememberedUser && (
        <>
          <ClickTracker name="create_account">
            <Button
              nohover
              type="button"
              link
              id={createAccount}
              className="create-account-header-label"
              onClick={e =>
                onLinkClick(
                  {
                    e,
                    openOverlay,
                    userNameClick,
                    triggerLoginCreateAccount,
                    navname: AnalyticsNavigationText.createAccount,
                  },
                  createAccount
                )
              }
              fontSizeVariation="large"
              anchorVariation="primary"
            >
              Create Account
            </Button>
          </ClickTracker>
          <ClickTracker name="log_in">
            <Button
              nohover
              type="button"
              link
              id={login}
              className="rightLink login-header-label"
              onClick={LoginLinkClick}
              fontSizeVariation="large"
              anchorVariation="primary"
            >
              Login
            </Button>
          </ClickTracker>
        </>
      )}

      {isRememberedUser && (
        <BodyCopy component="div" className="account-info-section" tabIndex="0">
          <ClickTracker name="log_in">
            <BodyCopy
              className="account-info user-name"
              component="div"
              role="button"
              id={login}
              onClick={LoginLinkClick}
            >
              {`Hi, ${userName}`}
            </BodyCopy>

            {!isDrawer && (
              <Image
                alt="user"
                src={getIconPath('down_arrow_icon')}
                height="6px"
                onClick={LoginLinkClick}
              />
            )}

            {!isCanada() ? (
              <BodyCopy component="div">
                <div className="account-info user-points">{`${userPoints} Points`}</div>
                <span className="account-info user-rewards rightLink">
                  {`$${handleUserRewards(userRewards)} Rewards`}
                </span>
              </BodyCopy>
            ) : null}
            {!isDrawer ? (
              <Image
                alt="user"
                className="usericon"
                src={getIconPath('user-icon')}
                onClick={LoginLinkClick}
              />
            ) : null}
          </ClickTracker>
        </BodyCopy>
      )}

      {!isDrawer ? (
        <Image
          alt="user"
          className="usericon"
          src={getIconPath('user-icon')}
          onClick={e =>
            onLinkClick({ e, openOverlay, userNameClick, triggerLoginCreateAccount }, login)
          }
        />
      ) : null}
    </React.Fragment>
  );
};

GuestUserInfo.propTypes = {
  createAccount: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  openOverlay: PropTypes.func.isRequired,
  userNameClick: PropTypes.bool.isRequired,
  onLinkClick: PropTypes.func.isRequired,
  triggerLoginCreateAccount: PropTypes.bool.isRequired,
  isDrawer: PropTypes.bool.isRequired,
  isRememberedUser: PropTypes.bool,
  userName: PropTypes.string,
  userPoints: PropTypes.string.isRequired,
  userRewards: PropTypes.string.isRequired,
};

GuestUserInfo.defaultProps = {
  isRememberedUser: false,
  userName: '',
};
export default GuestUserInfo;
