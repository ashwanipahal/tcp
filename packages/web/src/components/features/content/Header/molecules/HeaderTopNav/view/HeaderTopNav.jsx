import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { BodyCopy } from '@tcp/core/src/components/common/atoms';
import { routerPush } from '@tcp/core/src/utils';
import { BrandTabs, PromotionalArea } from '../..';
import HeaderTopNavStyle from '../HeaderTopNav.style';

const HeaderTopNav = ({
  className,
  brandTabs,
  promoMessageWrapper,
  openOverlay,
  isUserLoggedIn,
}) => {
  const onLinkClick = e => {
    e.preventDefault();
    if (!isUserLoggedIn) openOverlay({ state: true });
    else routerPush('/account', '/account');
  };
  return (
    <div className={className}>
      <PromotionalArea mobile data={promoMessageWrapper} />
      <div className="header-topnav__row">
        <div className="header-topnav__brand-tabs">
          <BrandTabs data={brandTabs} />
        </div>
        <div className="header-topnav__promo-area">
          <PromotionalArea mobile={false} data={promoMessageWrapper} />
        </div>
        <React.Fragment>
          <BodyCopy
            id="trackOrder"
            textAlign="right"
            className="header-topnav__track-order"
            onClick={e => onLinkClick(e)}
          >
            Track order
          </BodyCopy>
        </React.Fragment>
      </div>
    </div>
  );
};

HeaderTopNav.propTypes = {
  className: PropTypes.string.isRequired,
  brandTabs: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  promoMessageWrapper: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  openOverlay: PropTypes.func.isRequired,
  isUserLoggedIn: PropTypes.bool.isRequired,
};

export default withStyles(HeaderTopNav, HeaderTopNavStyle);
export { HeaderTopNav as HeaderTopNavVanilla };
