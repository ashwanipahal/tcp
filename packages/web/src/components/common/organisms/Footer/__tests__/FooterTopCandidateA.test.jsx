import React from 'react';
import { shallow } from 'enzyme';
import { FooterTopCandidateAVanilla } from '../views/FooterTopCandidateA.view';

describe('FooterTopCandidateAVanilla component', () => {
  it('renders correctly', () => {
    const component = shallow(<FooterTopCandidateAVanilla />);
    expect(component).toMatchSnapshot();
  });
});
