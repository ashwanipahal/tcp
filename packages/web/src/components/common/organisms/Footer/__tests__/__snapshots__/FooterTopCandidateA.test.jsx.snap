// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterTopCandidateAVanilla component renders correctly 1`] = `
<Styled(Grid)
  className="footer_top_candidate_a"
>
  <Styled(Row)>
    <Styled(Col)
      className=""
      colSize={
        Object {
          "large": 4,
          "medium": 4,
          "small": 6,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(BodyCopy)
        className="heading_text"
        color="text.primary"
        component="p"
        data-locator="email_promo_text"
        fontFamily="primary"
        fontSize={
          Array [
            "fs12",
            "fs12",
            "fs16",
          ]
        }
        fontWeight="black"
        letterSpacing="normal"
        lineHeight="normal"
        textAlign="center"
      >
        <Styled(RichText)
          richTextHtml="GET $10 OFF BY SIGNING UP FOR EMAIL!"
        />
      </Styled(BodyCopy)>
      <ReduxForm
        asyncBlurFields={
          Array [
            "signup",
          ]
        }
        asyncValidate={[Function]}
        dataLocators={
          Object {
            "inputField": "enter_email_text_field",
            "submitButton": "email_submit_btn",
          }
        }
        destroyOnUnmount={true}
        enableReinitialize={false}
        fieldName="signup"
        forceUnregisterOnUnmount={false}
        form="FooterTopEmailSignUpForm"
        getFormState={[Function]}
        initialValues={
          Object {
            "signup": "",
          }
        }
        keepDirtyOnReinitialize={false}
        labels={
          Object {
            "placeholderText": "Enter email address",
            "submitButtonLabel": "Submit",
            "termsTextLabel": "*Applies to new email subscribers only. Exclusions apply. Offer valid onyour next purchase of $40 or more. You may withdraw your consent at any time. Contact Us The Children’s Place, 500 Plaza Drive, Secaucus, NJ 07094, www.childrensplace.com.",
            "validationErrorLabel": "",
          }
        }
        onFormSubmit={[Function]}
        openSuccessModal={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        subscription={Object {}}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
      <Styled(BodyCopy)
        className={null}
        color="text.primary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": false,
              "rules": Array [
                "
  ",
                "
  ",
                [Function],
                ";
",
                ";
",
              ],
            },
            "displayName": "Styled(RichText)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        fontFamily="secondary"
        fontSize={
          Array [
            "fs12",
            "fs12",
            "fs14",
          ]
        }
        fontWeight="regular"
        letterSpacing="normal"
        lineHeight="normal"
        richTextHtml="*Applies to new email subscribers only. Exclusions apply. Offer valid onyour next purchase of $40 or more. You may withdraw your consent at any time. Contact Us The Children’s Place, 500 Plaza Drive, Secaucus, NJ 07094, www.childrensplace.com."
        textAlign="center"
      >
        <Styled(RichText)>
          *Applies to new email subscribers only. Exclusions apply. Offer valid onyour next purchase of $40 or more. You may withdraw your consent at any time. Contact Us The Children’s Place, 500 Plaza Drive, Secaucus, NJ 07094, www.childrensplace.com.
        </Styled(RichText)>
      </Styled(BodyCopy)>
      <div
        className="divider hide-in-medium-up"
      />
    </Styled(Col)>
    <Styled(Col)
      className=""
      colSize={
        Object {
          "large": 4,
          "medium": 4,
          "small": 6,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(BodyCopy)
        className="heading_text"
        color="text.primary"
        component="p"
        fontFamily="primary"
        fontSize={
          Array [
            "fs12",
            "fs12",
            "fs16",
          ]
        }
        fontWeight="black"
        letterSpacing="normal"
        lineHeight="normal"
        textAlign="center"
      >
        <Styled(RichText)
          richTextHtml="GET $10 OFF BY SIGNING UP FOR SMS!"
        />
      </Styled(BodyCopy)>
      <ReduxForm
        asyncBlurFields={
          Array [
            "footerTopSmsSignup",
          ]
        }
        asyncValidate={[Function]}
        destroyOnUnmount={true}
        enableReinitialize={false}
        fieldName="footerTopSmsSignup"
        fieldProps={
          Object {
            "normalize": [Function],
          }
        }
        forceUnregisterOnUnmount={false}
        form="FooterTopSmsSignUpForm"
        getFormState={[Function]}
        initialValues={
          Object {
            "footerTopSmsSignup": "",
          }
        }
        keepDirtyOnReinitialize={false}
        labels={
          Object {
            "placeholderText": "Enter phone number",
            "submitButtonLabel": "Submit",
            "termsTextLabel": "*Applies to new email subscribers only. Exclusions apply. Offer valid onyour next purchase of $40 or more. You may withdraw your consent at any time. Contact Us The Children’s Place, 500 Plaza Drive, Secaucus, NJ 07094, www.childrensplace.com.",
            "validationErrorLabel": "",
          }
        }
        onFormSubmit={[Function]}
        openSuccessModal={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        subscription={Object {}}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
      />
      <Styled(BodyCopy)
        className={null}
        color="text.primary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": false,
              "rules": Array [
                "
  ",
                "
  ",
                [Function],
                ";
",
                ";
",
              ],
            },
            "displayName": "Styled(RichText)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        fontFamily="secondary"
        fontSize={
          Array [
            "fs12",
            "fs12",
            "fs14",
          ]
        }
        fontWeight="regular"
        letterSpacing="normal"
        lineHeight="normal"
        richTextHtml="*Applies to new email subscribers only. Exclusions apply. Offer valid onyour next purchase of $40 or more. You may withdraw your consent at any time. Contact Us The Children’s Place, 500 Plaza Drive, Secaucus, NJ 07094, www.childrensplace.com."
        textAlign="center"
      >
        <Styled(RichText)>
          *Applies to new email subscribers only. Exclusions apply. Offer valid onyour next purchase of $40 or more. You may withdraw your consent at any time. Contact Us The Children’s Place, 500 Plaza Drive, Secaucus, NJ 07094, www.childrensplace.com.
        </Styled(RichText)>
      </Styled(BodyCopy)>
    </Styled(Col)>
    <div
      className="divider hide-in-large-up"
    />
    <Styled(Col)
      className="refer_a_friend_desktop"
      colSize={
        Object {
          "large": 4,
          "medium": 4,
          "small": 6,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(Grid)>
        <Styled(Row)
          className="flex-align-center"
          fullBleed={true}
        >
          <Styled(Col)
            className=""
            colSize={
              Object {
                "large": 7,
                "medium": 4,
                "small": 7,
              }
            }
            ignoreGutter={
              Object {
                "small": true,
              }
            }
          >
            <Styled(BodyCopy)
              className="heading_text refer_friend_text"
              color="text.primary"
              component="p"
              fontFamily="primary"
              fontSize={
                Array [
                  "fs12",
                  "fs12",
                  "fs16",
                ]
              }
              fontWeight="black"
              letterSpacing="normal"
              lineHeight="normal"
              textAlign="center"
            >
              <Styled(RichText)
                richTextHtml="$10 for a friend, $10 for you!"
              />
            </Styled(BodyCopy)>
          </Styled(Col)>
          <Styled(Col)
            className="candidate_a_inline_container_button"
            colSize={
              Object {
                "large": 5,
                "medium": 4,
                "small": 6,
              }
            }
            ignoreGutter={
              Object {
                "small": false,
              }
            }
          >
            <Styled(Button)
              buttonVariation="variable-width"
              data-locator="refer_a_btn"
              id="extole_zone_global_footer"
              type="refer_a"
            >
              REFER FRIEND
            </Styled(Button)>
          </Styled(Col)>
        </Styled(Row)>
        <div
          className="divider hide-in-medium-down"
        />
        <Styled(Row)
          className="hide-in-medium-down"
          fullBleed={true}
        >
          <Styled(Col)
            className=""
            colSize={
              Object {
                "large": 12,
                "medium": 12,
                "small": 12,
              }
            }
            ignoreGutter={
              Object {
                "small": true,
              }
            }
          >
            <div
              className="divider hide-in-medium-up"
            />
            <Styled(SocialMediaLinks)
              className="footer_top_candidate_a_social_links"
            />
          </Styled(Col)>
        </Styled(Row)>
      </Styled(Grid)>
    </Styled(Col)>
    <div
      className="divider hide-in-medium-up"
    />
    <Styled(Col)
      className="hide-in-large-up refer_a_frient_last_colm"
      colSize={
        Object {
          "large": 0,
          "medium": 4,
          "small": 12,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(SocialMediaLinks) />
    </Styled(Col)>
  </Styled(Row)>
  <div
    className="divider hide-in-medium-down"
  />
</Styled(Grid)>
`;
